src/Parsing/Impl/Ast/ImportStatement.cs

```diff
@@ -16,18 +16,17 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Python.Core.Collections;
 
 namespace Microsoft.Python.Parsing.Ast {
     public class ImportStatement : Statement {
-        private readonly ModuleName[] _names;
-        private readonly NameExpression[] _asNames;
-
-        public ImportStatement(ModuleName[] names, NameExpression[] asNames, bool forceAbsolute) {
-            _names = names;
-            _asNames = asNames;
+        public ImportStatement(ImmutableArray<ModuleName> names, ImmutableArray<NameExpression> asNames, bool forceAbsolute) {
+            Names = names;
```

Comment:
So they are guaranteed not be nulls?

Reply:
That's the idea. `ImmutableArray<T>` is a struct, so it can't be null. It can contain `null` elements. (see comment above). The only thing here is that `ImmutableArray<T>` must be created using `ImmutableArray<T>.Empty` or `ImmutableArray<T>.Create`. `default(ImmutableArray<T>)` will create uninitialized instance.